peps/pep-0817.rst

```diff
@@ -941,17 +1021,74 @@ the external plugin when the package in question is incompatible with
 the installer's implementation.
 
 
+Metadata in project tree and wheels
+-----------------------------------
+
+Variants introduce a few new portions of metadata that is stored in the
+project tree and in wheels. In the project tree, it is stored in the
+``pyproject.toml`` file along with other project properties, benefiting
+from the TOML format's readability and strictness. Afterwards, it is
+converted into an equivalent JSON structure, and stored as a separate
+file in the ``.dist-info`` directory. The existing metadata files are
+unchanged to avoid unnecessary incompatibility, and to avoid serializing
+into the inconvenient :doc:`Core Metadata
+<packaging:specifications/core-metadata>` format.
+
+The project tree-level metadata includes:
+
+- information about variant providers that could be used by the wheels,
+- optionally, lists overriding the default property ordering,
+- static property lists for Ahead-of-Time providers that do not use
+  plugins.
+
+In wheel metadata, the above is amended by static property lists
+obtained from the plugins and variant properties for the built wheel.
+
+When wheels are published on an index, the variant metadata from all
+wheels is combined into a single ``{name}-{version}-variants.json`` file
+that is used by clients to efficiently obtain the variant metadata
+without having to download it from individual wheels separately, or
+implement explicit variant metadata support on the package index
+software.
+
+
+ABI dependency variant provider
+-------------------------------
+
+Some packages provide extension modules exposing Application Binary
+Interface (ABI) that is not compatible across wide ranges of versions.
+The packages using this interface need to pin their wheels to the
+version used at build time. If ABI changes frequently, the pins are very
+narrow and users face problems if they need to install two packages that
+may happen to pin to different versions of the same dependency.
+Providing variants built against different dependency versions can
+increase the chances of resolver being able to find a dependency version
+that is compatible with all the packages being installed.
+
+Unfortunately, such a variant provider cannot be implemented within the
+plugin API defined by the specification. Given that a robust
+implementation would need to interface with the dependency resolver,
+rather than attempt to extend the API to cover this use case and add
+significant complexity as a result, the specification reserves
+``abi_dependency`` as a special variant namespace that can be
+implemented by installers wishing the provide this feature.
+
+Given the complexity of the problem, this extension is made entirely
+optional. This implies that any packages using it need to provide
+regular wheels as well.
```

Comment:
I suggest "regular" -> "non-variant" (or "regular (non-variant)"). I assume in a few years variant wheels _are_ regular wheels.